virtool/tasks/client.py

```diff
@@ -3,22 +3,32 @@
 
 from aioredis import Redis
 
+from virtool.tasks.pg import register
+
 
 class TasksClient:
 
-    def __init__(self, redis: Redis):
+    def __init__(self, redis: Redis, pg):
         self._redis = redis
+        self.pg = pg
 
-    async def add(self, task_id):
+    async def add(self, task_class, context=None):
         try:
-            await self._redis.publish("channel:tasks", task_id)
+            row = await register(self.pg, task_class, context=context)
+            await self._redis.publish("channel:tasks", row["id"])
+
+            return row
         except CancelledError:
             pass
 
-    async def add_periodic(self, task_id, interval):
+    async def add_periodic(self, task_class, interval, context=None):
         try:
             while True:
-                await self._redis.publish("channel:tasks", task_id)
+                row = await register(self.pg, task_class, context=context)
+
+                await self._redis.publish("channel:tasks", row["id"])
                 await asyncio.sleep(interval)
+
+                return row
         except asyncio.CancelledError:
             pass
```

Comment:
Docstrings and typing

Reply:
Changes already made in commit [2037eae](https://github.com/virtool/virtool/pull/2182/commits/2037eaefe5b49b20c2d0fdec71d1d30af7dda3ac)